packages/experimental/surface/src/plugins/RoutesPlugin.tsx

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { Dispatch, SetStateAction } from 'react';
+import { RouteObject, useRoutes } from 'react-router';
+import { HashRouter } from 'react-router-dom';
+
+import { definePlugin, Plugin, usePluginContext } from '../framework';
+
+export type ListViewContextValue = {
+  routes: RouteObject[];
+  setRoutes: Dispatch<SetStateAction<RouteObject[]>>;
+};
+
+export type RouterPluginProvides = {
+  router: {
+    routes: () => RouteObject[];
+  };
+};
+
+export type RouterPlugin = Plugin<RouterPluginProvides>;
+
+const routerPlugins = (plugins: Plugin[]): RouterPlugin[] => {
+  return (plugins as RouterPlugin[]).filter((p) => p.provides?.router);
```

Comment:
this feels unsafe and/or cumbersome? can we make the interface concrete/introspectable?
for inspiration: https://learn.microsoft.com/en-us/cpp/atl/queryinterface?view=msvc-170